src/error.rs

```diff
@@ -288,13 +288,7 @@ impl std::fmt::Debug for Error {
         let stack = self.stack();
 
         writeln!(f)?;
-        for (i, (_, source)) in stack.iter().skip(1).enumerate() {
-            match source {
-                Source::Root => {}
-                _ => writeln!(f, "    {}: {}", i, source)?,
-            }
-        }
-
+        write!(f, "{self:#}")?;
```

Comment:
are you sure this prints the same stack now? especially it doesn't skip the root anymore, so you will end up printing that twice now I think

Reply:
I was not getting the root at all before, in the new test I added. Maybe needs more testing, also with all the different enum types of Error (tested only the whatever variant so far, I think).

Reply:
btw: what is the difference between the `Source` and `Message` variants, couldn't they be merged?